core/etcd/config/templates/stack-template.json

```diff
@@ -348,24 +348,24 @@
           {{$etcdInstance.SubnetRef}}
         ]
       },
-      {{if $.WaitSignal.Enabled}}
+      {{if $.WaitSignal.Enabled -}}
       "CreationPolicy" : {
         "ResourceSignal" : {
           "Count" : "1",
-          "Timeout" : "{{$.Controller.CreateTimeout}}"
+          "Timeout" : "{{if $.EtcdMigrationEnabled}}PT3H{{else}}{{$.Controller.CreateTimeout}}{{end}}"
```

Comment:
Would you mind sharing me why it takes so much(3 hours)?!

Reply:
Hi, I haven't had one take that long yet, but I have experienced a small cluster going above the default 15 minutes.  The issue, I think, is with the import (exporting the keys seems performant enough).  The import script is a single threaded shell script that executes within an etcdctl container and I think that writing larger items, such as big configmaps, is slower than smaller keys.  I noticed that a number of helm related config maps were importing quite slowly.  I haven't tried to migrate any of our large cluster yet, but I am concerned that I may need to refactor the import to improve performance.